src/components/Locations.js

```diff
@@ -0,0 +1,98 @@
+import { Component } from "react";
+import "../App.css";
+
+class Locations extends Component{
+    constructor(){
+        super();
+
+        this.state={
+            locations:[],
+            display:true,
+            locationName:"",
+            locationClimate:"",
+            locationTerrain:"",
+            buttonText: "Show Locations"
+        }
```

Comment:
```suggestion
        this.state={
            locations:[],
            display:true,
        }
```
These are the only two pieces of state that you need. You can use conditional rendering that is **entirely** based on `this.state.display` to decide if the list of locations will be **shown** or **hidden**.